components/suggest/src/store.rs

```diff
@@ -3558,6 +3558,191 @@ mod tests {
+                  "icon": "2",
+                  "impression_url": "https://example.com/impression_url",
+                  "click_url": "https://example.com/click_url",
+                  "score": 0.3
+              }
+            ]),
+        )?
+        .with_icon("icon-3.png", "also-an-icon".as_bytes().into());
+
+        let store = unique_test_store(SnapshotSettingsClient::with_snapshot(snapshot));
+
+        store.ingest(SuggestIngestionConstraints::default())?;
+
+        let table = [
+            (
+                "keyword = `las`; Amp Mobile",
+                SuggestionQuery {
+                    keyword: "las".into(),
+                    providers: vec![SuggestionProvider::AmpMobile],
+                    limit: None,
+                },
+                expect![[r#"
+                [
+                    AmpMobile {
+                        title: "Mobile - Lasagna Come Out Tomorrow",
+                        url: "https://www.lasagna.restaurant",
+                        raw_url: "https://www.lasagna.restaurant",
+                        icon: None,
+                        full_keyword: "lasagna",
+                        block_id: 0,
+                        advertiser: "Good Place Eats",
+                        iab_category: "8 - Food & Drink",
+                        impression_url: "https://example.com/impression_url",
+                        click_url: "https://example.com/click_url",
+                        raw_click_url: "https://example.com/click_url",
+                        score: 0.3,
+                    },
+                ]
+                "#]],
+            ),
+            (
+                "keyword = `las`; Amp",
+                SuggestionQuery {
+                    keyword: "las".into(),
+                    providers: vec![SuggestionProvider::Amp],
+                    limit: None,
+                },
+                expect![[r#"
+                [
+                    Amp {
+                        title: "Desktop - Lasagna Come Out Tomorrow",
+                        url: "https://www.lasagna.restaurant",
+                        raw_url: "https://www.lasagna.restaurant",
+                        icon: None,
+                        full_keyword: "lasagna",
+                        block_id: 0,
+                        advertiser: "Good Place Eats",
+                        iab_category: "8 - Food & Drink",
+                        impression_url: "https://example.com/impression_url",
+                        click_url: "https://example.com/click_url",
+                        raw_click_url: "https://example.com/click_url",
+                        score: 0.3,
+                    },
+                ]
+                "#]],
+            ),
+            (
+                "keyword = `las `; amp and amp mobile",
+                SuggestionQuery {
+                    keyword: "las".into(),
+                    providers: vec![SuggestionProvider::Amp, SuggestionProvider::AmpMobile],
```

Comment:
Note that, in practice, consumers would only issue queries against either of those two providers, and it's up to the consumers to enforce that. But I guess having a test here to query both is fine for completeness.